src/goto-instrument/contracts/utils.cpp

```diff
@@ -682,6 +682,48 @@ get_loop_head(const unsigned int target_loop_number, goto_functiont &function)
   return get_loop_head_or_end(target_loop_number, function, true);
 }
 
+exprt get_loop_invariants(
+  const goto_programt::const_targett &loop_end,
+  const bool check_side_effect)
+{
+  auto invariant = static_cast<const exprt &>(
+    loop_end->condition().find(ID_C_spec_loop_invariant));
+  if(!invariant.is_nil() && check_side_effect)
+  {
+    if(has_subexpr(invariant, ID_side_effect))
```

Comment:
Can we separate the check for side-effect expressions from the accessor function ?

Reply:
I think we should have only one public function to extract loop contracts and check side-effect. But yes we can separate the accessor  functions as static private functions.